### PR TITLE
feat(ui): add password less related api

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add password less related api

## Linear Issue Reference

LOG-1790

## Testing

align with core/router
ut added
@logto-io/eng

### CODE REVIEW

File: packages/ui/src/apis/register.ts [hunk 1]
```diff
@@ -14,3 +14,53 @@ export const register = async (username: string, password: string) => {
...
+
+export const sendPhonePasscode = async (phone: string) => {
+  return ky
+    .post('/session/register/passwordless/phone/send-passcode', {
+      json: {
+        phone,
+      },
+    })
+    .json();
+};
```
Comment (demonzoo) [inline packages/ui/src/apis/register.ts:26]: Can we change the folder path from `apis` to `services`, as they are actually front-end service calls to call the backend APIs

Comment (simeng-li) [reply packages/ui/src/apis/register.ts:26]: That gonna be a cross ui project-wised update, prefer to update in a separate pr. 
Besides @logto-io/eng for alignment of the naming conversion, `api` or `services`.

Comment (wangsijie) [reply packages/ui/src/apis/register.ts:26]: Both are OK to me.

**Review by wangsijie — COMMENTED** [review]
Personally prefer to call `ky.post` directly in component's "submit" function.

**Review by wangsijie — APPROVED** [review]
Approve to unblock

**Comment by simeng-li** [comment]
> Personally prefer to call `ky.post` directly in component's "submit" function.

That is because we intend to have a useApi hook to wrap up all the API calls and provide a global loading/error/result handling logic.

File: packages/ui/src/apis/register.ts (hunk 1 above)
Comment (simeng-li) [reply packages/ui/src/apis/register.ts:26]: https://linear.app/silverhand/issue/LOG-2054/api-folder-name-convention

**Pull request merged**